crates/bifrost/src/providers/replicated_loglet/sequencer/appender.rs

```diff
@@ -279,6 +284,8 @@ impl<T: TransportConnect> SequencerAppender<T> {
             });
         }
 
+        // NOTE: It's very important to keep this look cancellation safe. If the appender future
+        // was cancelled, we don't want to move the global commit offset.
```

Comment:
Unrelated to this PR: When seeing a `StoreTaskStatus::Sealed` could we exit the `store_tasks.join_next()` early if we have reached fmajority?

Reply:
It only saves us, at best ~2s (the store timeout) but adds a f-majority check on every append which I'm not sure if it's worth the extra cost tbh. In the next PRs there will be some changes to this anyway to fix a few other issues.